src/libmongoc/tests/test-mongoc-dns.c

```diff
@@ -728,6 +728,117 @@ prose_test_9_pooled (void *unused)
    _prose_test_9 (true);
 }
 
+typedef struct {
+   int num_topology_opening;
+   int num_topology_closed;
+   int num_server_opening;
+   int num_server_closed;
+} cb_stats_t;
+
+static void
```

Comment:
A comment about why this name would be really helpful: is it creating an invalid state for purposes of testing? (It's not obvious that it is.)

Reply:
The names are intended to separate note these callbacks are applied to the `test_invalid_topology_pooled` and `test_invalid_topology_single` tests. Added a comment on the callbacks.

Reply:
+1 thanks-- naming things is hard!!